src/OpenApi/src/Extensions/OpenApiDocumentExtensions.cs

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.OpenApi.Models;
+using Microsoft.OpenApi.Models.References;
+
+namespace Microsoft.AspNetCore.OpenApi;
+
+internal static class OpenApiDocumentExtensions
+{
+    /// <summary>
+    /// Registers a <see cref="OpenApiSchema" /> into the top-level components store on the
+    /// <see cref="OpenApiDocument" /> and returns a resolvable reference to it.
+    /// </summary>
+    /// <param name="document">The <see cref="OpenApiDocument"/> to register the schema onto.</param>
+    /// <param name="schemaId">The ID that serves as the key for the schema in the schema store.</param>
+    /// <param name="schema">The <see cref="OpenApiSchema" /> to register into the document.</param>
+    /// <returns>An <see cref="OpenApiSchema"/> with a reference to the stored schema.</returns>
+    public static OpenApiSchema AddOpenApiSchemaByReference(this OpenApiDocument document, string schemaId, OpenApiSchema schema)
+    {
+        document.Components ??= new();
+        document.Components.Schemas ??= new Dictionary<string, OpenApiSchema>();
+        document.Components.Schemas[schemaId] = schema;
+        document.Workspace ??= new();
+        var location = document.BaseUri + "/components/schemas/" + schemaId;
```

Comment:
Is there any risk here of `BaseUri` ending with a `/` and messing up the concatenation?

Reply:
The BaseUri is currently implicitly configured by OpenAPI.NET ([ref](https://github.com/microsoft/OpenAPI.NET/blob/d9a2481483ec601f26dc0052bb488a75fb68a89d/src/Microsoft.OpenApi/Models/OpenApiDocument.cs#L105-L106)). The default value ends with a GUID and no trailing slash. They don't expose a `setter` for the property and I believe it's primarily there to have a base document URL for resolving schemas so I don't tink the risk is likely.